### ISSUE TITLE
Error by Debug of Testcase

### ISSUE THREAD

**Issue by AndreasRusjaev** [nokia/RED#166]
Hello,
i have Red 0.8.2, Python 2.7.14 win 32, RobotFramework 3.0.2.
If i had a BreakPoint in Testcase and start test tace in debug mode.
Red have no stop by breakbapont but i get a error:
![debugger-error](https://user-images.githubusercontent.com/18263592/34156485-c66f0ee4-e4bd-11e7-807a-dff7ef8c40b6.png)
and console write:
[console-output.txt](https://github.com/nokia/RED/files/1571585/console-output.txt)
My Red log:
[.log](https://github.com/nokia/RED/files/1571588/default.log)

**Comment by KrzysztofJozefowicz** [comment]
Thanks, we will look into it right away.

**Comment by AndreasRusjaev** [comment]
Thanks

**Comment by AndreasRusjaev** [comment]
i have java jdk 1.8.0_131

**Comment by KrzysztofJozefowicz** [comment]
Did you upgraded RED without restarting RED/Eclipse? It is one of the obvious cases since this is a basic function well tested in functional tests. If not, we would need testcase to reproduce it.

**Comment by AndreasRusjaev** [comment]
no i have new installatoin. I have on more as an one testcase error.

**Comment by AndreasRusjaev** [comment]
Hallo,
We use the 
robotframework-databaselibrary 1.0.1 and 
psycopg2 2.7.3.2(postgresqldriver) with 
PostreSql 9.6.5 
This error get after call ${qry_result}    Query    ${qry_string} 
${qry_string}  is one select query aka select * from foo.
I checked query wirh more as one select query the same error.

**Comment by michalanglart** [comment]
Ok,

thank you for details. We were able to reproduce it with this simple custom library:

`class database_lib(object):

    ROBOT_LIBRARY_VERSION = 1.0

    def __init__(self):
        pass

    def query(self):
        return [(1,2,3),(4,5,6)]`

and this suite:

`*** Settings ***
Library    database_lib    
*** Test Cases ***
case
    ${x}=    Query
    Log    ${x}`

The problem is with the way agent sending the tuples to RED when breakpoint is hit. It was introduced with implementation of #111 and released in 0.8.0. We'll fix that, but it looks like there is no simple workaround and the problem will occur when there are python tuples nested inside values of variables.

**Comment by AndreasRusjaev** [comment]
Thanks

**Comment by michalanglart** [comment]
Until we fixed that you may relatively simply patch this bug in your RED manually:

1) Unzip RED/plugins/org.robotframework.ide.eclipse.main.plugin_0.8.2.xxxxxxxxxxxx.jar
2) Unzip lib/org.robotframework.ide.core-functions-0.0.1-SNAPSHOT.jar 
3) Open scripts/TestRunnerAgent.py
4) modify _label_with_types() function by adding case for tuple:

```
def _label_with_types(data):
    value_type = type(data).__name__
    if isinstance(data, Mapping):
        return (value_type, dict((k, _label_with_types(data[k])) for k in data))
    elif isinstance(data, list):
        return (value_type, list(_label_with_types(el) for el in data))
    elif isinstance(data, tuple):
        return (value_type, tuple(list(_label_with_types(el) for el in data)))
    else:
        return (value_type, data)
```
5) store the modified file and zip everything back into jar files

**Comment by AndreasRusjaev** [comment]
Sorry,
a have made this changes, but the TestRunnerAgent.py in the temp directory was copied by run is without my changes.
What make i wrong?

**Comment by michalanglart** [comment]
Last time I checked it was working for me. Now I had troubles too: eclipse caches those data, so described way only works before first RED start. However adding `-clean` switch to `RED.ini` helped me. 

Remember to remove this switch later and keep in mind that this is dirty hack anyway which may potentially harm your RED/eclipse installation.

**Comment by AndreasRusjaev** [comment]
Thank You,
great thats work. 
I have tested it on my computer and in the computer of colleague.
Would you please insert this fix in the next release
thank you in advance

**Comment by michalanglart** [comment]
Yes, it will be definately included in next release

Glad to hear it does work eventually!

**Comment by KrzysztofJozefowicz** [comment]
released in 0.8.3
